### DISCUSSION TITLE
chapter_preface/about_the_book/

### DISCUSSION THREAD

**Discussion by giscus** [krahets/hello-algo#12] [Announcements]
# chapter_preface/about_the_book/

动画图解、一键运行的数据结构与算法教程

https://www.hello-algo.com/chapter_preface/about_the_book/

**Comment by NJUPT-Sivan** [comment]
请教一下作者的思维导图是用什么软件画的呀，感觉好好看。

**Reply by krahets** [reply]
哈喽，是用 Whimsical 画的哈

**Reply by NJUPT-Sivan** [reply]
谢谢大佬！

**Reply by hawkwean** [reply]
灰常感谢大佬的分享，打卡第一天，冲！

**Reply by Chelsea615** [reply]
nice!

**Comment by bigbitbox** [comment]
我来学习了！

**Reply by krahets** [reply]
来辽！

**Comment by youshaoXG** [comment]
先感谢博主，希望自己能有所收获！

**Reply by krahets** [reply]
冲冲冲！

**Comment by huangyfGG** [comment]
大佬，小弟我刚开始刷leetcode。我看了下本书的内容结构，想请问，如双指针，kmp，并集查询等，是不是不会像dp/贪心这样分专题讲，而是穿插到各个数据结构里分散讲，还是暂时没有规划？

**Reply by krahets** [reply]
哈喽，是暂时没有规划。相比于现有规划的内容，双指针的颗粒度较小，更像是一种解题范式技巧；kmp算法难度略高、不太适合初学者学习；并集查询是和数据库强相关的。

**Reply by huangyfGG** [reply]
哦哦，懂了。非常感谢大佬能百忙之中冲抽空回复我~

**Comment by ljyzzZ** [comment]
最近正是在看《图解算法数据结构》，原来是同一个作者！太棒了哈哈哈！

**Reply by krahets** [reply]
👀 这不是巧了吗哈哈

**Comment by guowei-gong** [comment]
leetcode 看到过来的，我就是在会与不会之间反复横跳的，现在做了快 200 题了，感觉看到困难题、中等题，都会畏惧的选手。还是没有开窍。

**Reply by krahets** [reply]
加油加油！会好起来的

**Comment by Zhuzhzzz** [comment]
太帅辣！

**Reply by krahets** [reply]
感谢鸭！

**Comment by duduxx123** [comment]
我这种刚入门的菜逼也看得懂，谢谢大佬

**Reply by krahets** [reply]
谢谢鼓励，加油冲！

**Comment by zxlhue** [comment]
大佬写的真的对我这个菜逼太合适了，简单易懂，开学考试都感觉不慌了（另外跪求图的更新doge）

**Reply by krahets** [reply]
谢谢鼓励！收到🫡，加足马力了！

**Comment by theLastWinner** [comment]
很酷！！！

**Reply by krahets** [reply]
谢谢嗷！

**Comment by vafyl** [comment]
刷leetcode的时候就看过k神的很多题解，新的一年要跳槽，赶紧算法刷起来！

**Reply by krahets** [reply]
刷起来，加油哇！

**Comment by Zhuzhzzz** [comment]
K神好！现在在学回溯，感到入门十分吃力。请问这本书还在更新嘛，会有相关内容吗

**Reply by krahets** [reply]
会的，回溯会在后续更新～加油！！

**Reply by BanDuWuZi** [reply]
动态规划呢？有计划吗

**Reply by krahets** [reply]
@BanDuWuZi 有的，快推出了～

**Comment by zh-h** [comment]
当年我面试情司，刚好会做全排列和查找，很有幸得以进入工作了好几年，最近某种原因分手了。
我得硬着头皮面试其他公司保住（提升）薪资，以后再回去🥹

**Reply by krahets** [reply]
冲！

**Comment by FangYuan33** [comment]
K神，有没有考虑加入像 “更新日志” 这样的章节，比如说我基本上把书看了一遍，而书再更新了新的内容时，之后再看我不太清楚哪里做了更新

**Reply by krahets** [reply]
谢谢建议！我在想给新增章节加一些标识

**Comment by JIFR2019** [comment]
这本书真是写的太好了，逻辑清晰，简单明了，配合各种图片能够快速理解数据结构和算法逻辑，比直接硬啃教材强多了。今年研二，下半年找工作需要刷算法题，这本书真的是及时雨！

**Reply by krahets** [reply]
谢谢鼓励！！

**Comment by Maukueng** [comment]
写的太好了，之前光刷算法题总感觉懵懵懂懂的，看完觉得对数据结构的理解清晰了很多，而且真正做到了深入又简明易懂，中午发现的宝藏网站，根本停不下来，晚上就看完了一遍，加入催更队伍。

**Reply by krahets** [reply]
谢谢鼓励！划重点：中午发现的、晚上就看完了。高效！！～

**Comment by ZhenhongDu** [comment]
为k神的开源精神点赞！
k神更新完了可以考虑整理成书，联系出版社进行出版，方便需要的读者，同时也能获得一些物质上的回报。

**Reply by krahets** [reply]
谢谢鼓励！等内容稳定下来整一整～最近也在搞 PDF 版，敬请期待！

**Comment by supercoko** [comment]
居然是K神, 之前就是在leetcode跟着k神的图解剑指offer刷的. 没想到又遇见了, 感谢k神的分享.

**Reply by krahets** [reply]
哇哈哈，再来温习一遍～

**Comment by keeprunni** [comment]
新手第一次使用vscode，但是发现中文乱码的问题，在代码里的中文注释是用utf-8的编码格式，但控制台输出是GBK的编码格式，打印出来后全部乱码了，完全看不懂内容。不知道怎么办？PS：把文件格式内容改成GBK，注释也全乱码了。。。

**Reply by krahets** [reply]
Hi ，文件保持在 utf-8 ，windows 的话可以试试这个 https://blog.csdn.net/u014302425/article/details/107908941 ，或者查一下其它解决方案

**Comment by ninja-liu** [comment]
可以考虑给一些专有名词添加上英文吗？

**Reply by krahets** [reply]
好呀，现在重要名词基本都有英文注释，后面我也打算提供一个中英文对照表

**Comment by BanDuWuZi** [comment]
大佬 我想问一下这个网页版的书 (https://www.hello-algo.com/) 是怎么制作的? 制作很精良，看起来很清晰，舒服。

**Reply by krahets** [reply]
是基于 Material-for-MkDocs 开发的哦～

**Comment by fallenleavesguy** [comment]
小时候总是看的pascal教材，又没互联网查资料。每次学到图就放弃了。这本书看起来很厉害，希望这次能坚持下去。感谢大佬

**Reply by krahets** [reply]
加油！

**Comment by myqianlan** [comment]
谢谢大佬

**Comment by 1024MrL** [comment]
感谢博主分享！鄙人有一拙见就是，如果用动画替换一些图解，小白理解起来会更为直观，比如排序算法这一部分

**Reply by krahets** [reply]
谢谢建议！确实。例如“选择排序”这一节的动画，你是否觉得比静态图更好理解？还是说你指的动画是 Gif 形式的那种？

**Reply by fallenleavesguy** [reply]
动画不能用自动播放的，要让人手动点下一步的，才好，要不不如现在这种分成1，2，3，4，5张图切换的

**Reply by krahets** [reply]
@fallenleavesguy 确实

**Comment by bethewind** [comment]
手机上giscus评论好像看不了,需要翻墙？

**Reply by WSL0809** [reply]
好像是，挂梯子秒加载

**Reply by krahets** [reply]
应该不需要吧

**Comment by daishuai** [comment]
好厉害

**Comment by FantasyLeo** [comment]
新读者+1

**Comment by ByYogi** [comment]
Hi, 能不能把章节切换的按钮优化固定(fixed)到窗口底部呢，每次点下一章都要经过评论划到网页最下方很不方便。

**Reply by krahets** [reply]
收到🫡，这个我看看如何实现

**Comment by lgl1227** [comment]
K神，你好，我隐隐约约记得早期的`hello 算法` ，每一章节存在思维导图来着，现在似乎没了

**Reply by krahets** [reply]
是的，有一张树状图，我觉得用处不大所以移除了。你需要看吗？

**Comment by lgl1227** [comment]
是那个目录的图吗？ 可以辛苦发一下嘛，谢谢

**Reply by krahets** [reply]
没问题，已在文中更新

**Reply by lgl1227** [reply]
谢谢大佬，另外我想问一下，这种树状图是用什么框架或者软件生成的

**Reply by krahets** [reply]
Whimsical

**Comment by lstar1314** [comment]
你好，这个网址是用什么现成的框架编写的么？左边的目录，中间的内容，右边的内容结构。我最近在编写一个在线文档手册，想学习下。

**Reply by krahets** [reply]
是现成框架，可以 github 搜下 mkdocs-material

**Reply by lstar1314** [reply]
好的，谢谢大佬能这么快速的回复

**Reply by lstar1314** [reply]
大佬，发现个问题，我在这里留言过，后续每次有人留言都会往我的邮箱发送邮件。这个是bug了么？或者可以怎么设置下不？

**Reply by SeaEpoch** [reply]
我也想知道😂……这样有点烦了！

**Comment by lgl1227** [comment]
👌谢谢

**Comment by ZQyang101** [comment]
感觉对小白很友好!

**Comment by quick1snail** [comment]
看了动态规划那一课总算是初级入门了，很好；可以整理出一个带目录的pdf版的吗？这样可以导入到pad里，边看边做笔记啥的

**Reply by krahets** [reply]
已有 PDF 版，在 GitHub 仓库里

**Comment by BluYous** [comment]
你好，为啥我本地跑起来没有右边的目录没有序号（x.x.x）呢？
> 目录
0.1.1   读者对象
0.1.2   内容结构
0.1.3   致谢

这是我本地跑的效果
> 目录
读者对象
内容结构
致谢

我翻遍了 https://squidfunk.github.io/mkdocs-material/ 好像也没看到有调节这个功能的开关。能指指路吗？

**Reply by krahets** [reply]
这个是我基于正则表达式写的一个简单的 "compiler" ，不在仓库里。

**Comment by xiaowu55** [comment]
先看的这本书，看完之后鼓起勇气去刷力扣。遇到不会不熟的知识点就回来复习，每次都看k神的题解，感觉图解的方式和作者很像，没想到就是本人。感谢分享hh

**Comment by qksuki** [comment]
图解通俗易懂，并且文章和图片的质量非常高！个人写文章时，图片的制作非常头疼，并没有相关的系统学习，无论是图片的样式、设计、编排在最后总是不能满意。这本书的图片设计非常精致，完全符合我心目中的高质量图片，心潮澎湃！请问下k神这类图片是使用什么软件来设计编排，并且是如何系统学习图片设计的？非常感谢！

**Reply by krahets** [reply]
用 ppt 或者 keynote 就可以画。多看看美的设计作品就好了

**Reply by qksuki** [reply]
谢谢🙏！

**Comment by yuan0221** [comment]
纪念下今天被 http://www.ruanyifeng.com/blog/2023/09/weekly-issue-270.html 推荐了

**Reply by krahets** [reply]
阮老师，YYDS！

**Reply by Shaoyi23** [reply]
感谢阮老师！找到宝藏了

**Comment by SeaEpoch** [comment]
一直想学好算法，但是一直很迷茫，这本书真的就是我的救命稻草😢！感谢大佬！
——打卡第一天2023.9.11

**Comment by dywtdance** [comment]
支持一下，同时也感谢阮老师的推荐

**Comment by mrsmallyi** [comment]
怎么没有PHP版本?嗷嗷,要不抽空我来搞个

**Reply by krahets** [reply]
暂时没有噢，欢迎来贡献 PHP 代码！可以看下 #15

**Comment by CodingGorit** [comment]
done

**Comment by Goodnameisfordoggy** [comment]
2023/9/13开始阅读

**Comment by fei19904287** [comment]
今天第一天

**Comment by 1z1y** [comment]
第一天

**Comment by Blue-Sky-F** [comment]
打卡阅读，内容很精美

**Comment by peterhefan** [comment]
1 点赞

**Comment by Lionlazy-Hao** [comment]
打卡，点赞

**Comment by lyyilin** [comment]
打卡，打卡，开始数据结构与算法之旅

**Comment by b-JwH-d** [comment]
第一条评论

**Comment by xtzyyyyy** [comment]
孤高的灵魂不需要羁绊之地！！！！

**Comment by jeffycai** [comment]
写的很通俗易懂，又很专业。

**Comment by leicheng42** [comment]
赞

**Comment by xiryg** [comment]
赞，加油

**Comment by imxiaomin** [comment]
talk is cheap, show me the code.

**Comment by Cos0vO** [comment]
START!!

**Comment by wjl1688** [comment]
START!

**Comment by gaoziman** [comment]
写的真好

**Comment by durk-web3** [comment]
太棒啦

**Comment by zhanghaoxing55** [comment]
大神 以后跟随

**Comment by zawaseinlove** [comment]
前来拜读

**Comment by S1-19027** [comment]
前来读一下
很后悔没有在高中学信息技术（选考）的时候看到这本书

**Comment by nonhana** [comment]
大三前端菜鸡前来拜读

**Comment by EEBOND7** [comment]
best wishes!

**Comment by Frankness1** [comment]
前人所荐，前来拜读。2023 10 23

**Comment by kianaaaaaa** [comment]
start

**Comment by lakerlu** [comment]
==

**Comment by heyangqing611** [comment]
泰酷辣

**Comment by chenb721** [comment]
跟风！

**Comment by cw123-hub** [comment]
帅

**Comment by ttggxx** [comment]
冲冲冲

**Comment by Itfuture03** [comment]
插个眼，开始，刷完再来回复自己

**Comment by xx10n31y** [comment]
时间复杂度评估是觉得最无聊的课程了, 希望能下定决心去学完这一章

**Comment by weiyuexin** [comment]
插眼

**Comment by Hanqianana** [comment]
good

**Comment by linksong22** [comment]
haha

**Comment by EricKiku** [comment]
好耶

**Comment by ROBIN-962301** [comment]
28岁 秃头计算机毕业笨逼 来此重新自学！

**Reply by krahets** [reply]
加油！！

**Comment by Ranpa369** [comment]
38岁红帽6架构师傅，温故知新😇

**Reply by krahets** [reply]
温故知新好啊！

**Comment by lkzwc** [comment]
才开始学呀

**Comment by kale2222** [comment]
人生迷茫期，给自己找点事做，慢慢行动起来。

**Comment by pxtzs** [comment]
给初学者一点建议谢谢

**Reply by krahets** [reply]
坚持学习～

**Comment by jackbai233** [comment]
pretty nice！！！字里行间感觉到作者的真诚，谦虚、认真。开始学习，感谢~

**Comment by stephenyoga** [comment]
oh 我前女友小名也叫泡泡

**Comment by muchengyang** [comment]
2024/01/23 坚持一个月后我在回来

**Comment by zhangzhangzhiyuan** [comment]
2024/1/25备战春招，冲冲冲

**Comment by iiiiiix111** [comment]
我看到许多算法书上有数学相关的证明，看完这本《hello算法》以后我还有必要去掌握数学相关的内容吗

**Reply by krahets** [reply]
看你的目标，想要深入研究算法可以多锻炼数学；如果是应用导向，则应该更倾向于理解原理和灵活运用

**Reply by zhuyihuan68** [reply]
有推荐的书籍吗

**Comment by zhangzhangzhiyuan** [comment]
看个人情况吧，hello算法算是入门，你要继续提升也最好遵循循序渐进，可以关注一下《代码随想录》有很多更高阶的玩法

猎魔人 ***@***.***> 于2024年1月27日周六 13:33写道：

**Comment by txlk62477** [comment]
感谢作者提供宝贵的资源

**Comment by Bertrand666** [comment]
大一备战蓝桥杯，冲冲冲，插个眼，等考完再回来看看！

**Comment by NingZiXi** [comment]
太赞了实在是

**Comment by newHopeLJP** [comment]
排版清新，阅读起来很舒服

**Comment by Rinato-C417** [comment]
Completed

**Comment by Feng-dll** [comment]
帅

**Comment by netjsg** [comment]
就是一个封面我都感觉清新脱俗，真的很不爽

**Comment by kslfkj** [comment]
打卡报道，一入算法深似海

**Comment by duduxx123** [comment]
好的，谢谢老师指导

**Comment by rlaoy** [comment]
来了

**Comment by yin-b** [comment]
不错的一本书

**Comment by pule150** [comment]
666

**Comment by chiacheng59** [comment]
謝謝作者用心整理

**Comment by jasontospas** [comment]
1111

**Comment by xueliii** [comment]
太棒了，感谢作者

**Comment by baixiangguo1001** [comment]
考研数据结构自命题能看这个学吗?

**Reply by krahets** [reply]
Hi，这本书不是针对考研的教辅书，但根据读者反馈，知识点重合度较高

**Comment by EnochGao** [comment]
最让人感动的是看到一堆感谢，这也许就是开源的魅力。

**Comment by 217heidai** [comment]
感谢各位大佬的辛苦付出

**Comment by panting666-afk** [comment]
感谢大佬们的付出！！

**Comment by Bingo333** [comment]
赞

**Comment by ding-wei** [comment]
先赞为敬!!!!!!!  加把劲希望自己也成为贡献者

**Comment by LHLHercules** [comment]
感谢各位大佬的努力与贡献

**Comment by yzx258** [comment]
点赞收藏，努力成为贡献者。

**Comment by zhaihuiliang** [comment]
由衷感谢作者，请问我可以在哪里对项目做赞助以表心意？

**Reply by krahets** [reply]
谢谢鼓励！本项目不接收赞助和投喂～如想支持可了解一下纸质书，祝阅读愉快！

**Reply by Wwh-SoEximious** [reply]
直接买纸质书，我已经入手了

**Comment by CoccaGuo** [comment]
🌱 ***CoccaGuo*** _2024年7月18日打卡_

**Comment by aizhiyuan** [comment]
加油

**Comment by CongChengC** [comment]
功德无量！

**Comment by tuoyuan2002** [comment]
🐂🍺

**Comment by Cocococoder** [comment]
来，开始学习算法了兄弟们

**Comment by Cali168Wei** [comment]
senior sde 回到美国的找工第一天，开始复习算法。。。

**Comment by laohugit** [comment]
k神这个在线文档用啥框架开发的，还是自己纯手工撸的？

**Reply by Harbour-z** [reply]
好像是mkdocs吧

**Reply by krahets** [reply]
Hi，是这个：[Material-for-MkDocs](https://github.com/squidfunk/mkdocs-material/tree/master)

**Comment by amufeng** [comment]
感受到了大佬的热忱，感谢😊

**Comment by KerwinCR** [comment]
之前一直偷懒逃避思考，算法学的不系统不扎实，很多知识点都似懂非懂，现在大三了，为了找工作慢慢开始认真刷力扣，无比痛苦，一做一个不吱声，也是因此看到了K神的题解，很喜欢K神，今天无意中接触到K神这么棒的开源分享项目，看了介绍的视频，更加佩服K神和K神等一众大佬了，让我第一次觉得互联网开源精神可以被诠释的这么生动，希望自己这次能不逃避，努力跟着K神好好学下来，享受算法的美

**Comment by robborta** [comment]
hello,world

**Comment by yanxingtong** [comment]
太棒啦

**Comment by jokerssssss9999** [comment]
大佬带带我

**Comment by Marigold-Lan** [comment]
太舒服了这个书看的

**Comment by yang123tian** [comment]
身为一名物联网的大四学生我很惭愧

**Comment by coronabear** [comment]
小白入门~

**Comment by NotUr77** [comment]
开始学习

**Comment by Shawnxzl** [comment]
图0-1线上的内容和下载的PDF不一样，下载的PDF说的都不是人话

**Reply by krahets** [reply]
请问哪里不一样？PDF 你是在官方下载的吗：https://github.com/krahets/hello-algo/releases

<img width="1338" alt="image" src="https://github.com/user-attachments/assets/e6a66229-4c46-4515-83ec-2ab0b1e9eca9" />

**Comment by Shawnxzl** [comment]
繁体和简体的区别，繁体的hash table翻译成离凑表有点

**Reply by krahets** [reply]
简体中文中的“哈希表”翻译到繁体中文就是“雜湊表”，请见 wiki：
https://zh.wikipedia.org/zh-tw/%E5%93%88%E5%B8%8C%E8%A1%A8

**Comment by sdyer-create** [comment]
厉害厉害

**Comment by WangYun0918** [comment]
图片中函数渐进上界错别字，应该是函数渐近上界

**Reply by krahets** [reply]
谢谢指正！

**Comment by pochapogo** [comment]
非常棒！
顺道问一下：请问这个网站前端部分用的什么框架？好漂亮！

**Reply by jidaojiuyou** [reply]
mkdocs

**Reply by jaywhj** [reply]
用的mkdocs-material，这个：

https://jaywhj.github.io/mkdocs-materialx

**Comment by JanHuber** [comment]
nice!!!
可以把仓库迁移到 gitcode 吗？ 学校里网络访问 github 网络受限，头大。

**Comment by quater4** [comment]
谢谢你的分享，谢谢所有帮助写这本书的人！

**Comment by Git52401314** [comment]
赶紧算法刷起来

**Comment by gongycn** [comment]
感谢分享，打卡

**Comment by Lvf-1936** [comment]
支持

**Comment by yangkegithub** [comment]
顶

**Comment by beginnerJq** [comment]
顶一个

**Comment by peidali266-gif** [comment]
顶

**Comment by GG220315** [comment]
太顶了，就是半路出家的同学们需要的

**Comment by jaywhj** [comment]
建议作者用最新的 materialx 更新下站点，更加美观不说，效率更高，速度更快啊，现在访问感觉响应有延迟，可以试试这个

https://jaywhj.github.io/mkdocs-materialx/

**Comment by Zay-074** [comment]
2026-1-5开始阅读

**Comment by 3124night** [comment]
窝腰盐书

**Reply by krahets** [reply]
书煤油温提

**Comment by Lct989P** [comment]
666

**Comment by limingwei666888-star** [comment]
2026.4.14 转达模型商科生第一天学习打卡~

**Comment by 0shu0** [comment]
要不说能成事的大佬后面都有坚强的后盾呢
